core/src/driver/stablex_driver.rs

```diff
@@ -118,9 +126,28 @@ impl<'a> StableXDriverImpl<'a> {
         };
 
         let submitted = if let Some(objective_value) = verified {
+            let gas_price_cap = {
+                let num_trades = solution.executed_orders.len();
+                match burnt_fees(&orders, &solution) {
```

Comment:
It's too bad that this method can fail (we could get rid of the hardcoded cap otherwise). Is this really a practical concern? I don't really see how a solution could be found for orders we don't have. But then asserting this here seems a bit far away from where this invariant is actually guaranteed. Maybe solution could have a burnt_fees field on it that we can populate when we get it and just read here?

Reply:
Some part of the code will have to do unwrap for this which I am ok with. I'll see if I find a more natural place for the calculation, that also has access to the original orders.

Reply:
I was under the impression that the solver result output contains burnt fees. Maybe we can reuse that (if it actually does)?

Reply:
Yes it does but I am unsure whether we should be relying on that information since the solver output format is not documented. If we use it then we say the solver always has to output it. But it's not much work to calculate on our side.

Reply:
Another way I thought of you make this method not fail is to just add together **all** sell amounts and subtract **all** buy amounts. I think you don't need to look at the order buy and sell tokens for this to work.

Additionally the BigInt to U256 conversion can be asserted or be saturating.

Reply:
Good idea. Implemented this.